### ISSUE TITLE
Check if esp32 workaround not needed ...

### ISSUE THREAD

**Issue by jepler** [adafruit/circuitpython#7030]
After #7023 maybe the `#if`'d workaround in **wifi_reset()** is not needed:
```c
void wifi_reset(void) {
    if (!wifi_inited) {
        return;
    }
    common_hal_wifi_monitor_deinit(MP_STATE_VM(wifi_monitor_singleton));
    wifi_radio_obj_t *radio = &common_hal_wifi_radio_obj;
    common_hal_wifi_radio_set_enabled(radio, false);
    #ifndef CONFIG_IDF_TARGET_ESP32
    ESP_ERROR_CHECK(esp_event_handler_instance_unregister(WIFI_EVENT,
        ESP_EVENT_ANY_ID,
        radio->handler_instance_all_wifi));
    ESP_ERROR_CHECK(esp_event_handler_instance_unregister(IP_EVENT,
        IP_EVENT_STA_GOT_IP,
        radio->handler_instance_got_ip));
    ESP_ERROR_CHECK(esp_wifi_deinit());
    esp_netif_destroy(radio->netif);
    radio->netif = NULL;
    esp_netif_destroy(radio->ap_netif);
    radio->ap_netif = NULL;
    wifi_inited = false;
    #endif
    supervisor_workflow_request_background();
}
```

**Comment by DavePutz** [comment]
I tested CP 8.1 beta on a doit_esp32_devkit_v1. After taking out the #ifdef in wifi_reset() in /ports/espressif/common-hal/wifi/__init__.c ;  I was able to run code.py multiple times and successfully reconnect to wifi. @jepler ,  are there some other tests that you would like done?

**Comment by jepler** [comment]
I don't have any devices with me to test, so please feel free to remove the workaround if your testing indicates it's not needed. I don't have any specific recollection, so I can only rely on what was in the related commit message:
> After this change, it's possible to connect multiple times to wifi in different runs of code.py or the repl after soft rebooting.

so it sounds like @DavePutz 's testing shows this is no longer needed.

I can't do any testing before May so I recommend going ahead and accepting a change related to this based on dave's testing.
